### PR TITLE
Make `proxy.forward.close-connection` configurable at all levels.

### PR DESCRIPTION
**Pull request by jhatala**

### CODE REVIEW

**Review by kazuho — COMMENTED** [review]
Thank you for the pull request. I like the idea of providing the knob at all levels.

File: include/h2o.h [hunk 1]
```diff
@@ -2173,6 +2169,7 @@ typedef struct st_h2o_proxy_config_vars_t {
     unsigned use_proxy_protocol : 1;
     unsigned tunnel_enabled : 1;
     unsigned connect_proxy_status_enabled : 1;
+    unsigned forward_close_connection : 1;
```
Comment (kazuho) [inline include/h2o.h:2172]: This looks good, I would appreciate us moving the doc-comment above as well, even when other properties do not have one.

File: include/h2o/httpclient.h [hunk 2]
```diff
@@ -155,6 +155,7 @@ typedef struct st_h2o_httpclient_ctx_t {
     size_t max_buffer_size;
     unsigned tunnel_enabled : 1;
     unsigned force_cleartext_http2 : 1;
+    unsigned forward_close_connection : 1;
```
Comment (kazuho) [inline include/h2o/httpclient.h:158]: I'm not sure if this is the correct place to add a flag that is not used by httpclient.

Considering that this flag is used by the proxy handler, maybe we can add the flag to `h2o_req_overrides_t` the same way like `use_proxy_protocol`.

**Review by kazuho — COMMENTED** [review]
Thank you for the pull request. I like the idea of providing the knob at all levels.

File: include/h2o.h (hunk 1 above)
Comment (jhatala) [reply include/h2o.h:2179]: That makes sense.  Done in a1d5cf0.

File: include/h2o/httpclient.h (hunk 2 above)
Comment (jhatala) [reply include/h2o/httpclient.h]: It works well that way, done in a1d5cf0, thank you!

**Comment by kazuho** [comment]
Thank you for the changes. Merged.

**Pull request merged**